http/rest-client-reactive/src/test/java/io/quarkus/ts/http/restclient/reactive/TLSRegistryIT.java

```diff
@@ -0,0 +1,221 @@
+import io.quarkus.ts.http.restclient.reactive.resources.TLSResource;
+import io.restassured.response.Response;
+
+@Tag("QUARKUS-5664")
+@QuarkusScenario
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class TLSRegistryIT {
+
+    @QuarkusApplication(ssl = true, properties = "tls.properties", classes = { MetaResource.class, TLSResource.class })
+    static final RestService server = new RestService()
+            .setAutoStart(false);
+
+    @QuarkusApplication(ssl = true, properties = "tls.properties", classes = { MetaClient.class, MetaClientResource.class,
+            TLSResource.class })
+    static final RestService client = new RestService()
+            .setAutoStart(false)
+            .withProperty("quarkus.rest-client.meta-client.url", () -> server.getURI(Protocol.HTTPS).toString());
+
+    private List<String> oldLogs = Collections.emptyList();
+
+    @Test
+    @Order(1)
+    // we need to check, that if trust-store.certificate-expiration-policy is not set, then it is "warn" by default
+    // other tests set this property for the "client app, so this method should be the first
+    public void testDefaultConnection() {
+        server.withProperty("quarkus.http.tls-configuration-name", "expired").start();
+        client.withProperty("quarkus.rest-client.meta-client.tls-configuration-name", "expired-client")
+                .start();
+
+        Response response = client.https().given().relaxedHTTPSValidation()
+                .get("/client/meta/headers");
+        Assertions.assertEquals(200, response.statusCode());
+        List<String> headers = response.body().jsonPath().getList(".");
+        assertTrue(headers.contains("User-Agent: [Quarkus REST Client]"),
+                "Response wasn't processed properly. The headers are:" + headers);
+        List<String> logs = client.getLogs();
+        logs.removeAll(oldLogs);
+
+        assertTrue(logContains(logs,
+                "A certificate has expired: java.security.cert.CertificateExpiredException"),
+                "Logs doesn't contain a warning about expired certificate! " + logs);
+    }
+
+    @Test
+    public void testValidConnection() {
+        server.withProperty("quarkus.http.tls-configuration-name", "server").start();
+        client
+                .withProperty("quarkus.rest-client.meta-client.tls-configuration-name", "client")
+                .withProperty("quarkus.tls.client.trust-store.certificate-expiration-policy", "reject")
+                .start();
+        Response response = client.https().given().relaxedHTTPSValidation()
+                .get("/client/meta/headers");
+        Assertions.assertEquals(200, response.statusCode());
+        List<String> headers = response.body().jsonPath().getList(".");
+        assertTrue(headers.contains("User-Agent: [Quarkus REST Client]"),
+                "Response wasn't processed properly. The headers are:" + headers);
+    }
+
+    @Test
+    public void testIgnoreConnection() {
```

Comment:
AFAICT this tests that expired client certificate is not rejected and there is no warning in the logs, so you can use the certificate to connect even though it is expired. Hence `testIgnoreConnection` does not describe what this test verifies.

Reply:
Something similar could be said about other test names, expect for the `testRejectedConnection` which is perfect.